src/util/string.cpp

```diff
@@ -423,198 +400,182 @@ static bool parseHexColorString(const std::string &value, video::SColor &color,
 	else
 		return false;
 
-	bool success = true;
-
 	for (size_t pos = 1, cc = 0; pos < len; pos++, cc++) {
-		assert(cc < sizeof components / sizeof components[0]);
 		if (short_form) {
-			unsigned char d;
-			if (!hex_digit_decode(value[pos], d)) {
-				success = false;
-				break;
-			}
+			u8 d;
+			if (!hex_digit_decode(value[pos], d))
+				return false;
+
 			components[cc] = (d & 0xf) << 4 | (d & 0xf);
 		} else {
-			unsigned char d1, d2;
+			u8 d1, d2;
 			if (!hex_digit_decode(value[pos], d1) ||
-					!hex_digit_decode(value[pos+1], d2)) {
-				success = false;
-				break;
-			}
+					!hex_digit_decode(value[pos+1], d2))
+				return false;
+
 			components[cc] = (d1 & 0xf) << 4 | (d2 & 0xf);
-			pos++;	// skip the second digit -- it's already used
+			pos++; // skip the second digit -- it's already used
 		}
 	}
 
-	if (success) {
-		color.setRed(components[0]);
-		color.setGreen(components[1]);
-		color.setBlue(components[2]);
-		color.setAlpha(components[3]);
-	}
+	color.setRed(components[0]);
+	color.setGreen(components[1]);
+	color.setBlue(components[2]);
+	color.setAlpha(components[3]);
 
-	return success;
+	return true;
 }
 
-struct ColorContainer {
-	ColorContainer();
-	std::map<const std::string, u32> colors;
+const std::unordered_map<std::string, u32> NAMED_COLORS = {
```

Comment:
```suggestion
const static std::unordered_map<const std::string, u32> NAMED_COLORS = {
```
not sure if the second `const` is needed here.

also despite being a constant the variable name should be lowercase

Reply:
I don't think a second const would do anything.